ui/components/src/Table/model/table-model.ts

```diff
@@ -59,6 +59,19 @@ export interface TableProps<TableData> {
    */
   checkboxSelection?: boolean;
 
+  /**
+   * Determines the behavior of row selection.
+   *
+   * - `standard`: clicking a checkbox will toggle that rows's selected/unselected
+   *    state and will not impact other rows.
+   * - `legend`: clicking a checkbox will "focus" that row by selecting it and
+   *    unselecting other rows. Clicking a checkbox with a modifier key pressed,
+   *    will change this behavior to behave like `standard`.
+   *
+   * @default 'standard'
+   */
+  rowSelectionVariant?: 'standard' | 'legend';
```

Comment:
I added a variant instead of just modifying the table behavior because the "legend" behavior is kind of weird for checkboxes and probably isn't want we want in non-legend table use cases.